src/GraphQL/Introspection/__Type.cs

```diff
@@ -49,8 +50,9 @@ public __Type()
                     }),
                 async context =>
                 {
-                    if (context.Source is IComplexGraphType type)
```

Comment:
@Shane32 I found bug. It was in that line. `InputObjectGraphType` also inherits from `IComplexGraphType` but it should not be here. So I really returned more fields.

Reply:
I suggest you add a test for this potential issue.

Reply:
#2233